02-peer-protocol.md

```diff
@@ -1166,12 +1178,14 @@ The sending node:
   next `commitment_signed` it expects to receive.
   - MUST set `next_remote_revocation_number` to the commitment number of the
   next `revoke_and_ack` message it expects to receive.
-  - if it supports `option_data_loss_protect`:
+  - if it supports `option_data_loss_protect` or `option_static_remotekey`:
     - if `next_remote_revocation_number` equals 0:
       - MUST set `your_last_per_commitment_secret` to all zeroes
     - otherwise:
       - MUST set `your_last_per_commitment_secret` to the last `per_commitment_secret`
     it received
+  - if `option_static_remotekey` applies to the commitment transaction:
+	- MUST NOT include `my_current_per_commitment_point`.
```

Comment:
does this mean that it should be blank?

Obviously haven't tried to implement it yet, but I'm wondering, do we have to modify this at all? Is it possible to keep all of the DLP handling/derivation the same, but just use the raw `payment_basepoint` in `to_remote` outputs?

Reply:
No, I want to kill it.  It's under-defined, it's been a source of problems, and even now I suspect we're seeing disagreements between c-lightning and lndL I'm seeing sync error far too often.

So, not blank, literally not present.  Of course, recipient will ignore anyway, but next extension may rely on this by appending a TLV...

Reply:
It still seems orthogonal to not tweaking the `to_remote` outputs. I can't imagine adding more logic in this area is going to improve the issues we're seeing... :)

Reply:
If we want to add additional fields to this in the future, then we can't have it be absent, since w/o TLV, we don't actually have "optional fields".

Reply:
Of course we can.  If we negotiate static_remotekey (which will eventually become compulsory), then the TLV goes immediately after it.  We won't support TLVs without static_remotekey I expect.

Reply:
> We won't support TLVs without static_remotekey I expect.

This _may_ be the case, but it seems short sighted to block of that path all together, when there's little cost to leaving the field as is. Newer implementations will simply write zeroes for that field, and then being the actual TLV parsing after that boundary. It's a fundamental trait that pre-TLV there're no optional fields, therefore we can't just "delete" them without adding additional parsing complexity. 

The other downside of "removing" this field, is that it makes parsing more contextual. This path forces parsing to be aware of the higher level context of which feature bits are online vs just reading the data, then interpreting it later.

Reply:
deleting it also a privacy leak, in that reveals what type of channel you have to anyone inspecting the size of your encrypted packets

Reply:
OK, as per meeting agreement, we'll leave it and ignore it.  http://www.erisian.com.au/meetbot/lightning-dev/2019/lightning-dev.2019-09-02-20.06.html

Reply:
I realized we may actually still need to _check_ it, even if we're using this new format. 

`lnd` has a recovery mode where in order to trigger the other party to force close if we have zero state transitions (no commit secret to check as nothing has been revoked), we'd purposefully set an invalid per commitment point. However, if the other party ignores it, then everything else looks fine: "state zero local, state zero remote, aight let's boot this thing up". So now Carol can't signal to Bob that they've lost data and Bob should force close.

Reply:
Put in less implementation specific terms: we need a way to signal that a party has lost data while they're on state zero, so the other person can force close. Previously `lnd` relied on the fact that we'd always check the commitment point. As a result, we'll (on our end implementation wise) will continue to always check that point. This isn't _mandatory_ persay, but would be nice if eclair/CL still checked that point to support this recovery case.